apps/crn-server/src/data-providers/reminders.data-provider.ts

```diff
@@ -32,6 +41,29 @@ export type ReminderDataProvider = DataProvider<
   FetchRemindersOptions
 >;
 
+type ResearchOutputQueried = Pick<
```

Comment:
Would be nice if this could get extracted but since this is deeply nested, I've found no way

Reply:
I don't think I understand why you should need it actually. You use it here:
https://github.com/yldio/asap-hub/blob/db2a41aeda360a5ebc8c6c4f314a45d4cea2eeb6/apps/crn-server/src/data-providers/reminders.data-provider.ts#L699
and
https://github.com/yldio/asap-hub/blob/db2a41aeda360a5ebc8c6c4f314a45d4cea2eeb6/apps/crn-server/src/data-providers/reminders.data-provider.ts#L732

The first of the two is then applied here
https://github.com/yldio/asap-hub/blob/db2a41aeda360a5ebc8c6c4f314a45d4cea2eeb6/apps/crn-server/src/data-providers/reminders.data-provider.ts#L327
when you follow the code from the reduce
https://github.com/yldio/asap-hub/blob/db2a41aeda360a5ebc8c6c4f314a45d4cea2eeb6/apps/crn-server/src/data-providers/reminders.data-provider.ts#L323

you can see that we reduce the `queryResearchOutputsContents` variable whose type is declared right above...
https://github.com/yldio/asap-hub/blob/db2a41aeda360a5ebc8c6c4f314a45d4cea2eeb6/apps/crn-server/src/data-providers/reminders.data-provider.ts#L307
...as `FetchReminderDataQuery['draftResearchOutputs']`; so your task is to simply extract the element of the array from that type. If you look at how we extract types in all data providers you should find that the way to achieve that is to via 
`NonNullable<FetchReminderDataQuery['draftResearchOutputs']>[number]`. So that means that your function declaration can be:
```typescript
const getAssociationNameAndType = (
  researchOutput: NonNullable<FetchReminderDataQuery['draftResearchOutputs']>[number],
): {
(...)
```

Reply:
We could wrap by a `NonNullable` and to get an item from the array we use this `[number]`. So it would be something like

```
type DraftResearchOutput = NonNullable<
  FetchReminderDataQuery['draftResearchOutputs']
>[number];
```